kafka-client/src/main/java/dev/responsive/kafka/api/config/ResponsiveConfig.java

```diff
@@ -231,6 +231,15 @@ public class ResponsiveConfig extends AbstractConfig {
       + "runs in. When set to RUN, runs the Kafka Streams app. When set to MIGRATE, runs app"
       + " migration.";
 
+  // ------------------ Misc functional overrides ----------------------
+  public static final String RESTORE_OFFSET_REPAIR_ENABLED_CONFIG = "responsive.restore.offset.repair.enabled";
+  public static final boolean RESTORE_OFFSET_REPAIR_ENABLED_DEFAULT = false;
+  public static final String RESTORE_OFFSET_REPAIR_ENABLED_DOC = "When set to 'true', " + RESTORE_OFFSET_REPAIR_ENABLED_CONFIG
+      + " will ignore OffsetOutOfRangeException and instead seek to the earliest available offset. This exception "
+      + "should only happen in situations where there is truncation/retention on the changelog topic and restoring from the latest "
+      + "committed offset in the remote store is no longer possible. Note that in some situations this may cause data "
```

Comment:
I'm still not really understanding where it can actually cause data loss.

It depends on what the source of truth is I guess. If the changelog is the source of truth, it should not be truncated by responsive, and any truncation in the changelog is like deleting that data from the store so its safe to skip.

If we are treating the remote store as the source of truth, then responsive could truncate the changelog and we should be more careful.

Today, we usually operate in the first mode. I think over time we want to move to the second mode. I wonder if it makes sense to make that distinction explicit as a config. Not really a comment on this PR in particular, but more spitballing - maybe we should have the config be between these two modes of operation and then decide to skip or not skip accordingly.

Reply:
Let's chat about it more offline. I agree with your logic, but something still makes me anxious.